### DISCUSSION TITLE
entry.client.tsx Env Vars

### DISCUSSION THREAD

**Discussion by jcpaulco** [remix-run/remix#4048] [Q&A] [answered]
I'm trying to access environment vars inside of entry.client.tsx and cannot figure out how this works.. Any tips would be greatly appreciated.

**Comment by sergiodxa** [comment, marked as answer]
Environment variables are only available server-side, there's no env variables on browsers, if you want to access your server env variables on the browser you would have to expose them from a loader.

You can read more about this and some approaches on this page of the docs https://remix.run/docs/en/v1/guides/envvars#browser-environment-variables

**Reply by jcpaulco** [reply]
Right that makes sense but is there a way to acccess them from the client entry point? `entry.client.tsx`?

**Reply by sergiodxa** [reply]
If you follow the guide on the docs, they suggest storing them in `window.ENV`, and because the script doing that is placed before the Remix scripts you can do `window.ENV.something` on `entry.client.tsx`

**Reply by jcpaulco** [reply]
Ahh i see now, thank you!
